src/lib.rs

```diff
@@ -747,6 +747,11 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
     marker: marker::PhantomData<(&'a K, &'a mut V)>,
 }
 
+/// A consuming insertion-order iterator over a `LinkedHashMap`'s entries.
+pub struct IntoIter<K, V, S> {
```

Comment:
You can make the iterator contain the head and tail of the list directly
instead of having it contain the map as a whole.

On Friday, April 22, 2016, David Tolnay <notifications@github.com> wrote:

> In src/lib.rs
> <https://github.com/contain-rs/linked-hash-map/pull/50#discussion_r60802506>
> :
>
> > @@ -747,6 +747,11 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
> >      marker: marker::PhantomData<(&'a K, &'a mut V)>,
> >  }
> >
> > +/// A consuming insertion-order iterator over a `LinkedHashMap`'s entries.
> > +pub struct IntoIter<K, V, S> {
>
> If we can't mention S, we can't keep the LinkedHashMap. How about
> collecting the values into a vector and iterating that?
>
> HashMap gets around this by keeping the private RawTable that backs the
> HashMap.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/contain-rs/linked-hash-map/pull/50/files/6bf364b7bc09322a7f900536f1e78bd8d03904be#r60802506>
>